src/main/java/seedu/address/model/Wallet.java

```diff
@@ -0,0 +1,159 @@
+package seedu.address.model;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.Objects;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.transaction.Expense;
+import seedu.address.model.transaction.Income;
+import seedu.address.model.transaction.Transaction;
+import seedu.address.model.transaction.TransactionList;
+
+
+/**
+ * Wraps all data at the wallet level
+ */
+public class Wallet implements ReadOnlyWallet {
+
+    private final TransactionList<Income> incomes = new TransactionList<>();
+    private final TransactionList<Expense> expenses = new TransactionList<>();
+
+    public Wallet() {}
+
+    /**
+     * Creates an Wallet using the Transactions in the {@code toBeCopied}
+     */
+    public Wallet(ReadOnlyWallet toBeCopied) {
+        this();
+        resetData(toBeCopied);
+    }
+
+    //// list overwrite operations
+
+    /**
+     * Replaces the contents of the income list with {@code incomes}.
+     */
+    public void setIncomes(List<Income> incomes) {
+        this.incomes.setTransactions(incomes);
+    }
+
+    /**
+     * Replaces the contents of the expense list with {@code expenses}.
+     */
+    public void setExpenses(List<Expense> expenses) {
+        this.expenses.setTransactions(expenses);
+    }
+
+    /**
+     * Resets the existing data of this {@code Wallet} with {@code newData}.
+     */
+    public void resetData(ReadOnlyWallet newData) {
+        requireNonNull(newData);
+        setIncomes(newData.getIncomeList());
+        setExpenses(newData.getExpenseList());
+    }
+
+    //// income-level operations
+
+    /**
+     * Returns true if an identical income exists in the Wallet
+     */
+    public boolean hasIncome(Income income) {
+        requireNonNull(income);
+        return incomes.contains(income);
+    }
+
+    /**
+     * Adds an income to the Wallet.
+     */
+    public void addIncome(Income income) {
+        incomes.add(income);
+    }
+
+    /**
+     * Replaces the given income {@code target} in the list with {@code editedIncome}.
+     * {@code target} must exist in the Wallet
+     */
+    public void setIncome(Income target, Income editedIncome) {
+        requireNonNull(editedIncome);
+        incomes.setTransaction(target, editedIncome);
+    }
+
+    /**
+     * Removes {@code key} from this {@code Wallet}.
+     * {@code key} must exist in the wallet.
+     */
+    public void removeIncome(Income key) {
+        incomes.remove(key);
+    }
+
+    //// expense-level operations
```

Comment:
I think you can try to make your section dividers consistent

Reply:
👌